docs/ota-client-guide/modules/ROOT/pages/virtual-secondaries.adoc

```diff
@@ -0,0 +1,55 @@
+= Use of Virtual Secondaries
+ifdef::env-github[]
+
+[NOTE]
+====
+We recommend that you link:https://docs.ota.here.com/ota-client/latest/{docname}.html[view this article in our documentation portal]. Not all of our articles render correctly in GitHub.
+====
+endif::[]
+
+
+Virtual secondary devices can be used as a way to update a configuration file on the primary device (where the primary device is a standard OSTree image running an aktualizr). You can do this in two parts:
+
+* First you need a virtual secondary configured that can be used to update a configuration file.
+* Afterwards you can integrate it into your Yocto build.
+
+
+
+== Creating a Virtual Secondary
+
+. You can create a virtual secondary to run locally by following the steps on how to xref:simulate-device-basic.adoc[simulate a device without building a disk image].
++
+You can use the provided `.deb` releases mentioned and the `virtualsec.json` file, which you can edit to suit your needs: such as removing one of the two entries and setting the paths and names to whatever you prefer.
+. Afterwards, you will be able to download the arbitrary files to the location specified in the "firmware_path" variable, by uploading the files to the https://connect.ota.here.com[OTA Connect Portal].

```

Comment:
```suggestion
. Afterwards, you will be able to download the arbitrary files to the location specified in the `firmware_path` variable, by uploading the files to the https://connect.ota.here.com[OTA Connect Portal].
```
Is this step necessary, though? As it is, we mention uploading twice, and this sentence seems just a bit confusing to me. Maybe it should be a higher-level comment.

Reply:
Should I remove this step about downloading files to that file path and only mention uploading to the OTA Connect portal?

Reply:
I think it's good to mention the part about downloading to that path, but I'd suggest moving it. I think it's just that mentioning the uploading and downloading in one sentence suggests that this happens simultaneously, when in fact they are two separate steps.

I see the steps of the process (at a high level) as something like this:
1. Configure the virtual secondary (which could include the bit about the download path).
2. Provision the device with the secondary.
3. Upload files for the secondary to the web UI.
4. Issuing the update/campaign.
5. Let the device download the file, which will probably happen automatically, but should be verified if desired or for demonstration purposes. (This might also include the bit about the path.)